chia/full_node/mempool_check_conditions.py

```diff
@@ -132,40 +134,222 @@ def mempool_assert_my_amount(condition: ConditionWithArgs, unspent: CoinRecord)
+
+
+def parse_condition_args(args: SExp, condition: ConditionOpcode) -> Tuple[int, Optional[List[bytes]]]:
+    """
+    Parse a list with exactly the expected args, given opcode,
+    from an SExp into a list of bytes. If there are fewer or more elements in
+    the list, raise a RuntimeError. If the condition is inherently true (such as
+    a time- or height lock with a negative time or height, the returned list is None
+    """
+    if condition is ConditionOpcode.AGG_SIG_UNSAFE or condition is ConditionOpcode.AGG_SIG_ME:
+        return ConditionCost.AGG_SIG.value, parse_aggsig(args)
+    elif condition is ConditionOpcode.CREATE_COIN:
+        return ConditionCost.CREATE_COIN.value, parse_create_coin(args)
+    elif condition is ConditionOpcode.ASSERT_SECONDS_ABSOLUTE:
+        return ConditionCost.ASSERT_SECONDS_ABSOLUTE.value, parse_seconds(args, Err.ASSERT_SECONDS_ABSOLUTE_FAILED)
+    elif condition is ConditionOpcode.ASSERT_SECONDS_RELATIVE:
+        return ConditionCost.ASSERT_SECONDS_RELATIVE.value, parse_seconds(args, Err.ASSERT_SECONDS_RELATIVE_FAILED)
+    elif condition is ConditionOpcode.ASSERT_HEIGHT_ABSOLUTE:
+        return ConditionCost.ASSERT_HEIGHT_ABSOLUTE.value, parse_height(args, Err.ASSERT_HEIGHT_ABSOLUTE_FAILED)
+    elif condition is ConditionOpcode.ASSERT_HEIGHT_RELATIVE:
+        return ConditionCost.ASSERT_HEIGHT_RELATIVE.value, parse_height(args, Err.ASSERT_HEIGHT_RELATIVE_FAILED)
+    elif condition is ConditionOpcode.ASSERT_MY_COIN_ID:
+        return ConditionCost.ASSERT_MY_COIN_ID.value, parse_coin_id(args, Err.ASSERT_MY_COIN_ID_FAILED)
+    elif condition is ConditionOpcode.RESERVE_FEE:
+        return ConditionCost.RESERVE_FEE.value, parse_fee(args)
+    elif condition is ConditionOpcode.CREATE_COIN_ANNOUNCEMENT:
+        return ConditionCost.CREATE_COIN_ANNOUNCEMENT.value, parse_announcement(args)
+    elif condition is ConditionOpcode.ASSERT_COIN_ANNOUNCEMENT:
+        return ConditionCost.ASSERT_COIN_ANNOUNCEMENT.value, parse_hash(args, Err.ASSERT_ANNOUNCE_CONSUMED_FAILED)
+    elif condition is ConditionOpcode.CREATE_PUZZLE_ANNOUNCEMENT:
+        return ConditionCost.CREATE_PUZZLE_ANNOUNCEMENT.value, parse_announcement(args)
+    elif condition is ConditionOpcode.ASSERT_PUZZLE_ANNOUNCEMENT:
+        return ConditionCost.ASSERT_PUZZLE_ANNOUNCEMENT.value, parse_hash(args, Err.ASSERT_ANNOUNCE_CONSUMED_FAILED)
+    elif condition is ConditionOpcode.ASSERT_MY_PARENT_ID:
+        return ConditionCost.ASSERT_MY_PARENT_ID.value, parse_coin_id(args, Err.ASSERT_MY_PARENT_ID_FAILED)
+    elif condition is ConditionOpcode.ASSERT_MY_PUZZLEHASH:
+        return ConditionCost.ASSERT_MY_PUZZLEHASH.value, parse_hash(args, Err.ASSERT_MY_PUZZLEHASH_FAILED)
+    elif condition is ConditionOpcode.ASSERT_MY_AMOUNT:
+        return ConditionCost.ASSERT_MY_AMOUNT.value, parse_amount(args)
+    else:
+        raise ValidationError(Err.INVALID_CONDITION)
+
+
+opcodes: Set[bytes] = set(item.value for item in ConditionOpcode)
```

Comment:
Maybe use all-caps (more pythonic to use upper case for module-scoped globals). Maybe call it `CONDITION_OPCODES` to differentiate between clvm opcodes.